### DISCUSSION TITLE
Is the installation of specific revision possible?

### DISCUSSION THREAD

**Discussion by dokimyj** [Homebrew/discussions#2249] [Everyday usage] [answered]
I'm using Homebrew/brew via cloning master branch for CI Mac OS Image (catalina/big sur)

and I was using `brew extract --version 8.4p1 --force openssh openssh/openssh-portable` for the image creation.

Since 2wks ago (actually after ldns 1.7.1 rev4 released in Homebrew/homebrew-core) this stopped working.

It shows

```
11:53:45 ==> ./configure --prefix=/<Image home path>/homebrew/Cellar/openssh@8.4p1
11:53:45 Last 15 lines from /<Image home path>/Library/Logs/Homebrew/openssh@8.4p1/01.configure:
11:53:45 checking for stdlib.h... (cached) yes
11:53:45 checking for GNU libc compatible malloc... yes
11:53:45 checking for stdlib.h... (cached) yes
11:53:45 checking for GNU libc compatible realloc... yes
11:53:45 checking if calloc(0, N) returns non-null... yes
11:53:45 checking for GLOB_ALTDIRFUNC support... yes
11:53:45 checking for gl_matchc field in glob_t... yes
11:53:45 checking for gl_statv and GLOB_KEEPSTAT extensions for glob... no
11:53:45 checking whether GLOB_NOMATCH is declared... yes
11:53:45 checking whether VIS_ALL is declared... no
11:53:45 checking whether struct dirent allocates space for d_name... yes
11:53:45 checking for /proc/pid/fd directory... no
11:53:45 checking for ldns-config... /<Image home path>/homebrew/opt/ldns/bin/ldns-config
11:53:45 checking for ldns support... no
11:53:45 configure: error: ** Incomplete or missing ldns libraries.
```
Same thing happens for the latest version extract (openssh@8.8p1)

Both modules have latest releases in 2019.

There is no install fails in typical `brew install openssh` in the Mac which I used for image creation,
so I think I might need ldns 1.7.1_3 installation before installing openssh@8.4p1.

Is there any way to extract specific revision of the module?

**Comment by carlocab** [comment, marked as answer]
Nothing built-in, unfortunately. You can just `brew extract` ldns 1.7.1 then modify your extracted formula so that it is equivalent to the state of the original formula at revision 3.

`revision`s reflect changes only in the formula itself, so you should be able to replicate the revision you want in the new formula you've extracted.

**Reply by dokimyj** [reply]
Thanks for your answer.

(Tried modifying formula, but unfortunately it seems that the page  prior revision of `ldns` does not exist anymore...)

```
16:54:40 ==> Downloading https://ghcr.io/v2/homebrew/core/ldns/1.7.1/manifests/1.7.1_3
16:54:40 curl: (22) The requested URL returned error: 404 
16:54:40 Error: ldns@1.7.1: Failed to download resource "ldns@1.7.1_bottle_manifest"
16:54:40 Download failed: https://ghcr.io/v2/homebrew/core/ldns/1.7.1/manifests/1.7.1_3
```
